tests/php/json-api/test-class.json-api-jetpack-endpoints.php

```diff
@@ -104,4 +217,65 @@ public function test_get_term_feed_url_ugly_permalinks() {
 			$response->feed_url
 		);
 	}
+
+	/**
+	 * Data provider for test_allows_site_based_authentication.
+	 */
+	public function data_provider_test_allows_site_based_authentication() {
+		return array(
+			'allow_jetpack_site_auth: true; logged_in_user: false;'  => array( true, false, true ),
+			'allow_jetpack_site_auth: false; logged_in_user: false;' => array( false, false, false ),
+			'allow_jetpack_site_auth: true; logged_in_user: true;'   => array( true, true, false ),
+		);
+	}
+
+	/**
+	 * Data provider for test_private_site_accessibility.
+	 */
+	public function data_provider_test_private_site_accessibility() {
+		$success = 'success';
+		$error   = new WP_Error( 'unauthorized', 'User cannot access this private blog.', 403 );
+
+		return array(
+			'allow_jetpack_site_auth: true; blog_token: true; can_read: null'   => array( true, true, null, $success ),
+			'allow_jetpack_site_auth: false; blog_token: true; can_read: null'   => array( false, true, null, $error ),
+			'allow_jetpack_site_auth: false; blog_token: false; can_read: false'   => array( false, false, false, $error ),
+			'allow_jetpack_site_auth: false; blog_token: false; can_read: true'   => array( false, false, true, $success ),
+		);
+	}
+
+	/**
+	 * Data provider for test_endpoint_capabilities.
+	 */
+	public function data_provider_test_endpoint_capabilities() {
+		$success = 'success';
+		$error   = new WP_Error( 'unauthorized', 'This user is not authorized to manage_options on this blog.', 403 );
```

Comment:
Isn't checking the code enough? If we change or fix something in the error message in the future tests will break

Reply:
That's the plan :)